library/kani/src/lib.rs

```diff
@@ -326,3 +326,7 @@ pub use kani_macros::*;
 pub(crate) use kani_macros::unstable_feature as unstable;
 
 pub mod contracts;
+
+pub fn apply_closure<T, U: FnOnce(&T) -> bool>(f: U, x: &T) -> bool {
```

Comment:
We should use `Fn` since the closure should not mutate the state.

Reply:
Also, please move this function to the `internal` module, and add documentation that this should only be used by contracts.

Maybe create an expected test where this call can fail, for example, mutating expression to see what the error will look like. Thanks!

Reply:
We usually mark internal functions with `#[doc(hidden)]`. However, I don't know if that is the case if they show up in the error message